### ISSUE TITLE
Removing Contracts and Filters

### ISSUE THREAD

**Issue by irr** [datacenter/acitoolkit#117]
Hi,
 
  As shown in aci_demo_contract.py, I was able to create a new contract and setup several filters as well. When I try to remove a contract or filter, using mark_as_deleted() and pushing to APIC, everything work as usual but in the "Secutity Policies -> Filters" the property created with <contract name> + <entry name> is not removed. See sample code bellow:

```python
from acitoolkit.acitoolkit import *
import requests, requests.packages
import acitoolkit

requests.packages.urllib3.disable_warnings()

APIC_USER = '...'
APIC_PWD = '...'

session = Session("https://apic", APIC_USER, APIC_PWD)
session.login()

tenant = Tenant.get_deep(session, names=('test',))[0]

contract = Contract('mysql-contract', tenant)
entry = FilterEntry('mysql',
                     applyToFrag='no',
                     arpOpc='unspecified',
                     dFromPort='3306',
                     dToPort='3306',
                     etherT='ip',
                     prot='tcp',
                     sFromPort='1',
                     sToPort='65535',
                     tcpRules='unspecified',
                     parent=contract)

print(tenant.push_to_apic(session))
print("Contract created.")

contract.mark_as_deleted()

print(tenant.push_to_apic(session))
print("Contract removed.")
```

After that, inside Secutity Policies -> Filters I still get 'mysql-contractmysql' (see bellow)
![mysql-contractmysql](https://cloud.githubusercontent.com/assets/178839/9330330/0e635de8-458f-11e5-9105-39a457781f34.jpg)

What should I do to remove all the entries (filters) related to old contract?

Thanks in advance,

   Ivan Rocha

**Comment by dnq** [comment]
You can modify the code to this:
```
print(tenant.push_to_apic(session))
print("Contract created.")

entry1.mark_as_deleted()
contract.mark_as_deleted()

print(tenant.push_to_apic(session))
print("Contract removed.")
```
This will remove the Filter Entry, but not the Filter itself.

It looks like this is related to my issue (#116). I'm not sure if there is currently a way to delete a Filter.

**Comment by irr** [comment]
Hi Richy,

   Yes, I already did it in my code (entry.mark_as_deleted() - I forgot to put it in my sample code) but, as you said, it will remove the Filter Entry and not the Filter itself. In my case, using acitoolkit for automation I really need to remove those unused Filters. 

   Let's wait for any help (from both https://github.com/datacenter/acitoolkit/issues/116 and https://github.com/datacenter/acitoolkit/issues/117) because I also could not find a way to delete a Filter using acitoolkit.

**Comment by dnq** [comment]
Hi Ivan,

I made a pull request that implements the 2 required classes (#118)
You can already try the code for yourself through my fork (https://github.com/dnq/acitoolkit), would be great to get some feedback from you :-)

Regards
Richy

**Comment by dnq** [comment]
Example Code of new Functionality:
```
#!/usr/bin/env python3
from acitoolkit.acitoolkit import *

session = Session('https://IP', 'User', 'PW')
session.login()

t = Tenant('API-Test-Tenant')

def getObject(obj_name):
    obj_search = Search()
    obj_search.name = obj_name
    return t.find(obj_search)[0]

for _ in range(5):
    con_name = 'API_Contract_' + str(_)
    con = Contract(con_name, t)
    sub_name = 'API_Contract_' + str(_) + '_Subject'
    sub = ContractSubject(sub_name, con)
    filt_name = 'API_filter_' + str(_)
    filt = Filter(filt_name, sub)
    entry_name = 'API_ssh_' + str(_)
    entry = FilterEntry(entry_name, filt, dFromPort='22', dToPort='22', etherT='ip', prot='tcp')

t.push_to_apic(session)

# Snipe some single Objects
getObject('API_ssh_4').mark_as_deleted()
getObject('API_filter_3').mark_as_deleted()
getObject('API_Contract_2_Subject').mark_as_deleted()
getObject('API_Contract_1').mark_as_deleted()

t.push_to_apic(session)

# Cleanup
# t.mark_as_deleted()
# t.push_to_apic(session)
```

**Comment by irr** [comment]
Hi Richy,

   Good news. I had to do a rollback and change some parts of my codebase because I had already replaced some code from acitoolkit to cobra sdk to deal with filters and then I was abled to test your patch. Everything worked as supposed with the new code (https://github.com/datacenter/acitoolkit/pull/118). Congratulations! Thanks a lot.
   
    Now everything is working in the same manner as using the APIC GUI.

   Regards,

      Ivan
